node/src/main/scala/coop/rchain/node/configuration/Configuration.scala

```diff
@@ -34,9 +34,11 @@ object Configuration {
   private val CertificateConfigKey = s"${hocon.Tls.Key}.${hocon.Tls.keys.Certificate}"
   private val KeyConfigKey         = s"${hocon.Tls.Key}.${hocon.Tls.keys.Key}"
   private val GenesisPathConfigKey = s"${hocon.Casper.Key}.${hocon.Casper.keys.GenesisPath}"
+  private val PrivateKeyPathConfigKey =
+    s"${hocon.Casper.Key}.${hocon.Casper.keys.ValidatorPrivateKey}"
```

Comment:
Variable here says key **path**, but from casper config you are using  `ValidatorPrivateKey` and not `ValidatorPrivateKeyPath`.

Reply:
The configuration is a `final case class` and the `privateKey`  is stored in conf. Once we build the configuration we can not change it.
I actually ask for the password while building the configuration. 
If the `ValidatorPrivateKeyPath ` is set, decrypt it and then set the `ValidatorPrivateKey` in the config. Otherwise, it would read straight from `ValidatorPrivateKey` option.
The `ValidatorPrivateKey` option is hidden which only should be used for the tests currently.
It is not an elegant way to implement it, I agree.